### PR TITLE
Local archive date and count

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Thanks to MIP11, values for count and date of archive (or fall back to description with filename) is available.

See https://github.com/Clinical-Genomics/MIP/pull/1763.

Screenshot 2022-06-22 at 17 48 20

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by CR, DN

### CODE REVIEW

File: scout/adapter/mongo/variant_loader.py [hunk 1]
```diff
@@ -379,6 +380,8 @@ def _load_variants(
             category(str): ['snv','sv','cancer','str']
             sample_info(dict): A dictionary with info about samples.
                                Strictly for cancer to tell which is tumor
+           custom_images(dict): A dict with custom images for a case.
```
Comment (northwestwitch) [inline scout/adapter/mongo/variant_loader.py:383]: 💯

File: scout/parse/variant/headers.py [hunk 2]
```diff
@@ -23,6 +23,42 @@ def parse_rank_results_header(vcf_obj):
     return rank_results_header
 
 
+def parse_local_archive_header(vcf_obj):
+    """Return a dict with local archive data for the case.
+
+    Check if the rank result is in the vcf header. If it exists return a dict
```
Comment (northwestwitch) [inline scout/parse/variant/headers.py:29]: Isn't this the loqusdb stats and not the rank results?

File: scout/parse/variant/headers.py [hunk 3]
```diff
@@ -23,6 +23,42 @@ def parse_rank_results_header(vcf_obj):
...
+    for header_line in vcf_obj.raw_header.split("\n"):
+        if "Software" in header_line and "loqusdb" in header_line:
+            software_entry = header_line.split("<")[1]
+            entries = software_entry.split(",")
+            for entry in entries:
+                assignment = entry.split("=")
+                the_key = assignment[0]
+                if the_key == "Date":
+                    the_value = assignment[1]
+                    local_archive_header["Date"] = the_value
```
Comment (northwestwitch) [inline scout/parse/variant/headers.py:57]: 👍🏻

**Review by northwestwitch — APPROVED** [review]
Good job, we can finally get rid of the old issue of the hard coded (wrong date) on the old obs panel! 🥇 🥳 

Works nicely (tested on the demo instance), We could perhaps try to load one real MIP11 case later as well.

I've noticed that in the code you interchange the `local_archive_info` and `local_archive_header` terms. I know it's the same thing, since the info is captured from the header (it works slightly different from rank_results_header for instance, where the header is further used to capture values from the variant lines). But might be confusing, Why don't you add a comment in the code saying just that?

You could also add a test. Should be easy now that the variants VCF contains a MIP11-like header no?

File: scout/parse/variant/headers.py (hunk 2 above)
Comment (dnil) [reply scout/parse/variant/headers.py:29]: Indeed! 😆

**Comment by northwestwitch** [comment]
Tested on Hasta server by uploading  a recent case [lovinggoat](https://scout-stage.scilifelab.se/cust000/NA24143-200ng/variants?variant_type=clinical).

The local observation doesn't change from the same case loaded on prod server. Isn't this case a MIP 11 case?

**Comment by dnil** [comment]
> Tested on Hasta server by uploading a recent case [lovinggoat](https://scout-stage.scilifelab.se/cust000/NA24143-200ng/variants?variant_type=clinical).
> 
> The local observations date doesn't change from the same case loaded on prod server. Isn't this case a MIP 11 case?
> 
> Mmm I can't find the "Software" and "loqusdb" strings in the SNVs header..

lovinggoat now on stage:

Screenshot 2022-06-23 at 12 20 10

The locusdb Software tag is indeed missing, ping @jemten.

**Comment by dnil** [comment]
Oki, merging with what we have. The Software loqus tag parsing works on local and test.

**Pull request merged**